pcdsdevices/smarpod.py

```diff
@@ -0,0 +1,266 @@
+"""
+Module for the SmarPod and related devices.
+"""
+from ophyd.device import Component as Cpt
+from ophyd.device import Device
+from ophyd.signal import EpicsSignal
+from .interface import BaseInterface
+
+
+
+class SmarPod(BaseInterface, Device):
```

Comment:
This class will not play nicely with bluesky for the purposes of scanning as you've implemented it as a generic `ophyd.Device`.

There needs to be a `Positioner` somewhere for the smarpod.

More details here: https://confluence.slac.stanford.edu/display/PCDS/How+to+Create+an+Ophyd+Device#HowtoCreateanOphydDevice-PVPositioner

Consider the upstream source of this SmarPod IOC implementation and its corresponding ophyd device: https://github.com/NSLS-II-HXN/profile_collection/blob/d724c469b8c4ac1784ca755b40b1992b3e5d0640/startup/15-zp.py#L15-L38 (and further down where it's being used as a sub-device)

Reply:
Interesting, it looks like NSLS-II splits this kind of device into many different 1D positioners

Reply:
PVPositioner was chosen as the class.
from ophyd import PVPositioner